### PR TITLE
XAPTC-43 Get All Test Cases

### PR DESCRIPTION
**Pull request by frankquinn**
Function, units tests and godog tests for Get all test cases

### CODE REVIEW

File: uiServices/src/testCaseServices.go [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+}
+
+func TestCaseCtx(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		next.ServeHTTP(w, r)
+	})
+
+}
+func getTestCaseHeader(req *http.Request) string {
+	testCasePathDir := req.Header.Get("testCasePathDir")
```
Comment (jenny-flynn) [inline uiServices/src/testCaseServices.go]: make this a constant

File: uiServices/src/testCaseServices.go [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+	for _, file := range files {
+		if filepath.Ext(testCasePathDir +file.Name()) == ".xml" {
+
+			testCase := new(testStrategies.TestDefinition)
+
+			filename := file.Name()
+
+			file, err := os.Open(fmt.Sprintf("%s%s", testCasePathDir, filename))
+			if err != nil {
+				logrus.Error("Cannot open file: ", filename)
```
Comment (jenny-flynn) [inline uiServices/src/testCaseServices.go]: for any of these errors, you should skip the file with ```continue```

File: uiServices/src/testCaseServices_test.go [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+
+func TestSuccessfulGetAllCases(t *testing.T) {
+	r := chi.NewRouter()
+	r.Mount("/", GetIndexPage())
+
+	DirectoryPath := os.Getenv("GOPATH") + "/src/github.com/xtracdev/automated-perf-test/uiServices/test/"
+	request, err := http.NewRequest(http.MethodGet, "/test-cases", nil)
+
+	request.Header.Set("testCasePathDir", DirectoryPath)
```
Comment (jenny-flynn) [inline uiServices/src/testCaseServices_test.go]: make these constants

File: uiServices/src/testCaseServices_test.go [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+	request.Header.Get("testCasePathDir")
+
+	w := httptest.NewRecorder()
+	r.ServeHTTP(w, request)
+
+	if err != nil {
+		t.Error(err)
```
Comment (jenny-flynn) [inline uiServices/src/testCaseServices_test.go]: you can replace this with 
```
if assert.NoError(err) { 
assert.Equal(t, http.StatusOK, w.Code, "Did not get all test cases")
}
```

Comment (jenny-flynn) [reply uiServices/src/testCaseServices_test.go]: Should also have a tests checking the number of test cases returned and the response type e.g. can it be unmarshalled into []Case from the json

File: uiServices/test/testCase.feature [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+  As an API user
+  I want to be able use various requests for test cases
+  So that I can test my application using custom metrics
+
+
+                                ###################################
+                                #######    GET ALL REQUESTS #######
+                                ###################################
+
+  Scenario: Susscessful retrieval all test cases with valid "GET" request
```
Comment (jenny-flynn) [inline uiServices/test/testCase.feature]: Successful

File: uiServices/src/testCaseServices.go [hunk 6]
```diff
@@ -60,18 +61,17 @@ func getAllTestCases(rw http.ResponseWriter, req *http.Request) {
 
 			file, err := os.Open(fmt.Sprintf("%s%s", testCasePathDir, filename))
 			if err != nil {
-				logrus.Error("Cannot open file: ", filename)
```
Comment (jenny-flynn) [inline uiServices/src/testCaseServices.go]: do log the error, but afterwards, continue

**Comment by jenny-flynn** [comment]
Resolve conflicts

**Comment by frankquinn** [comment]
Resolved Conflicts, ready for merge

**Pull request merged**